src/g1.rs

```diff
@@ -442,6 +438,46 @@ impl G1Affine {
         // y^2 - x^3 ?= 4
         (self.y.square() - (self.x.square() * self.x)).ct_eq(&B) | self.infinity
     }
+
+    /// Adds two affine points together.
+    /// In the zkvm context, this is accelerated with precompiles. In regular rust, this entails
+    /// converting one of the points to projective coordinates and then converting the output back.
+    pub fn add_affine(&self, rhs: &Self) -> Self {
+        if self.is_identity().into() {
+            return rhs.clone();
+        } else if rhs.is_identity().into() {
+            return self.clone();
+        }
+
+        cfg_if::cfg_if! {
+            if #[cfg(all(target_os = "zkvm", target_vendor = "succinct"))] {
+                let mut res = self.clone();
+                res.x.mul_r_inv_internal();
+                res.y.mul_r_inv_internal();
+                // The add precompile only works when P != Q
```

Comment:
@wwared how do you account for the affine formula also *not* working (and leading to a division by zero) when -P == Q?